### PR TITLE
V14: External login providers

### PR DESCRIPTION
**Pull request by iOvergaard**
## Description

Updated frontend code examples for external login providers in the reference article as well as the tutorial "Add google authentication".

## Type of suggestion

* [x] Updated outdated content

## Product & version (if relevant)
14.0.0

## Deadline (if relevant)

Before release on May 30th

### CODE REVIEW

**Comment by alina-tincas** [comment]
Hi @iOvergaard thank you for the PR 🙌 

We will review and merge it as soon as possible 🙏

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 1]
```diff
@@ -241,24 +241,59 @@ public class GenericBackOfficeExternalLoginProviderOptions : IConfigureNamedOpti
...
+        },
+        "linking": {
+          "allowManualLinking": true
+        }
+      }
+    }
+  ]
+}
+```
+
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:279]:
```suggestion
{% endcode %}
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 2]
```diff
@@ -241,24 +241,59 @@ public class GenericBackOfficeExternalLoginProviderOptions : IConfigureNamedOpti
...
+You have a few options to configure the button:
 
-If you want to use a custom icon for the login button, you need to add the icon to the Umbraco backoffice. You can do this by adding the icon to the `~/App_Plugins/MyPlugin/BackOffice/Icons` folder. The icon should be a SVG file. The icon should be named the same as the icon name you specify in the `options.Icon` property.
+* `element` - Define your own custom element for the button. This is useful if you want to display something other than a button, e.g. a link or an image. See the [Customizing the BackOffice login button](external-login-providers.md#customizing-the-backoffice-login-button) section for more information.
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:282]:
```suggestion
* `element` - Define your own custom element for the button. This is useful if you want to display something other than a button, For example: a link or an image. For more information, see the [Customizing the BackOffice login button](external-login-providers.md#customizing-the-backoffice-login-button) section.
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 3]
```diff
@@ -241,24 +241,59 @@ public class GenericBackOfficeExternalLoginProviderOptions : IConfigureNamedOpti
...
+* `forProviderName` - The name of the provider you are configuring. This should match the `SchemeName` in the `GenericBackOfficeExternalLoginProviderOptions` class with "Umbraco." prepended.
+* `meta.label` - The label to display on the button. This is what the user will see, e.g. "Sign in with Generic".
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:284]:
```suggestion
* `meta.label` - The label to display on the button. The user will see this text. For example: "Sign in with Generic".
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 4]
```diff
@@ -241,24 +241,59 @@ public class GenericBackOfficeExternalLoginProviderOptions : IConfigureNamedOpti
...
+* `meta.defaultView.look` - (Default: "default") The look of the button. You can use any of the looks from the [Umbraco UI Library](https://uui.umbraco.com/?path=/story/uui-button--looks-and-colors).
+* `meta.behavior.autoRedirect` - Automatically redirects the user to the external login provider, skipping the Umbraco login page, unless the user has specifically logged out or timed out.
+* `meta.behavior.popupTarget` - (Default: "umbracoAuthPopup") The target for the popup window. This is the name of the window that will be opened when the user clicks the button. If you want to open the login page in a new tab, you can set this to "_blank".
+* `meta.behavior.popupFeatures` - (Default: "width=600,height=600,menubar=no,location=no,resizable=yes,scrollbars=yes,status=no,toolbar=no") The features of the popup window. This is a string of comma-separated key-value pairs, e.g. "width=600,height=600". You can read more on the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#features).
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:290]:
```suggestion
* `meta.behavior.popupFeatures` - (Default: "width=600,height=600,menubar=no,location=no,resizable=yes,scrollbars=yes,status=no,toolbar=no") The features of the popup window. This is a string of comma-separated key-value pairs. For example: "width=600,height=600". You can read more on the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#features).
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 5]
```diff
@@ -241,24 +241,59 @@ public class GenericBackOfficeExternalLoginProviderOptions : IConfigureNamedOpti
...
+* `meta.linking.allowManualLinking` - Allows the user to link/unlink their account from the BackOffice. You need to allow manual linking on the `ExternalSignInAutoLinkOptions` as well.
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:291]:
```suggestion
* `meta.linking.allowManualLinking` - Allows the user to link orunlink their account from the BackOffice. You need to allow manual linking on the `ExternalSignInAutoLinkOptions` as well.
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 6]
```diff
@@ -241,24 +241,59 @@ public class GenericBackOfficeExternalLoginProviderOptions : IConfigureNamedOpti
...
+
+The button will now be displayed on the login page in the Umbraco backoffice.
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:293]:
```suggestion
The button will now be displayed on the login page in the Umbraco Backoffice.
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 7]
```diff
@@ -694,3 +572,187 @@ public static class ProviderMemberAuthenticationExtensions
 {% endtabs %}
 
 For a more in-depth article on how to set up OAuth providers in .NET refer to the [Microsoft Documentation](https://learn.microsoft.com/en-us/aspnet/core/security/authentication/social/?view=aspnetcore-8.0\&tabs=visual-studio).
+
+### Customizing the BackOffice login button
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:576]:
```suggestion
### Customizing the BackOffice Login Button
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 8]
```diff
@@ -694,3 +572,187 @@ public static class ProviderMemberAuthenticationExtensions
...
+
+If you want to customize the login button, you can do so by adding a custom element to the manifest file. This is useful if you want to display something other than a button, e.g. a link or an image.
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:578]:
```suggestion
If you want to customize the login button, you can do so by adding a custom element to the manifest file. This is useful if you want to display something other than a button. For example, a link or an image.
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 9]
```diff
@@ -694,3 +572,187 @@ public static class ProviderMemberAuthenticationExtensions
...
+
+The path to the custom view is a virtual path, like this example: `"~/App_Plugins/MyPlugin/BackOffice/my-external-login.js"`.
+
+When a custom view is specified it is 100% up to this module to perform all required logic.
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:582]:
```suggestion
When a custom view is specified, it is 100% up to this module to perform all the required logic.
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 10]
```diff
@@ -694,3 +572,187 @@ public static class ProviderMemberAuthenticationExtensions
...
+
+The module should define a Custom Element and export it as default. The Custom Element can optionally declare a number of properties to be passed to it. These properties are:
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:584]:
```suggestion
The module should define a Custom Element and export it as default. Optionally, the Custom Element can declare a number of properties to be passed to it. These properties are:
```

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 11]
```diff
@@ -694,3 +572,187 @@ public static class ProviderMemberAuthenticationExtensions
...
+{% endcode %}
+{% endtab %}
+
+{% tab title="Lit (JavaScript)" %}
+It is also possible to use a library like [Lit](https://lit.dev/) to render the custom element. The following example shows how to use Lit to render the custom element. The custom element will render a form with a button. The button will submit the form to the `externalLoginUrl` property. We do not have to perform any logic in the `constructor` method because Lit will automatically update any event listeners. Styling is also handled by Lit in the `static styles` property.
+
+We are using Lit version 3 in this example imported directly from a JavaScript delivery network to keep the example slim. You can also use a bundler like [Vite](https://vitejs.dev) to bundle the Lit library with your custom element.
+
+{% hint style="info" %}
+Check out the tutorial [Creating your first extension](../../tutorials/creating-your-first-extension.md) to learn more about how to set up a project with Vite.
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:708]:
```suggestion
To learn more about how to set up a project with Vite, see the [Creating your first extension](../../tutorials/creating-your-first-extension.md) tutorial.
```

**Review by eshanrnh — APPROVED** [review]
Looks good to me 🙌

File: 14/umbraco-cms/reference/security/external-login-providers.md [hunk 12]
```diff
@@ -241,24 +241,59 @@ public class GenericBackOfficeExternalLoginProviderOptions : IConfigureNamedOpti
...
+      }
     }
+  ]
 }
 ```
 {% endcode %}
+You have a few options to configure the button:
 
-**Icons**
+* `element` - Define your own custom element for the button. This is useful if you want to display something other than a button, For example: a link or an image. For more information, see the [Customizing the BackOffice login button](external-login-providers.md#customizing-the-backoffice-login-button) section.
```
Comment (eshanrnh) [inline 14/umbraco-cms/reference/security/external-login-providers.md:282]:
```suggestion
* `element` - Define your own custom element for the button. This is useful if you want to display something other than a button, For example: a link or an image. For more information, see the [Customizing the BackOffice Login Button](external-login-providers.md#customizing-the-backoffice-login-button) section.
```

**Pull request merged**